code/modules/ai/action_states/action_state.dm

```diff
@@ -0,0 +1,57 @@
+//A singleton action state; AI components are added to the list and a subsystem will periodically call the process() for this
+/datum/element/action_state
+
+/datum/element/action_state/New()
+	..()
+	if(src)
+		SSactionstate.processing += src
+
+/datum/element/action_state/Attach(mob/living/carbon/mob)
+	if(mob && istype(mob))
+		..()
+	else
+		return ELEMENT_INCOMPATIBLE
+
+//A special process() that doesn't rely on the processing subsystem but rather the action_states subsystem
+/datum/element/action_state/proc/state_process(mob/living/carbon/mob)
+
+/datum/element/action_state/move_to_atom
+	var/list/distances_to_maintain = list() //Distance we want to maintain from atom and send signals once distance has been maintained
+	var/list/atoms_to_walk_to = list()
+	var/list/last_moves = list() //Last world.time we moved at, move delay calculation
```

Comment:
This should be a var on the mob if it isn't already

Reply:
last_move_time is the variable on atom/movable, adjusted code